test/rateLimiters/tokenBucket.test.ts

```diff
@@ -0,0 +1,81 @@
+import TokenBucket from '../../src/rateLimiters/tokenBucket';
+
+const CAPACITY = 10;
+const REFILL_RATE = 1; // 1 token per second
+
+let limiter: TokenBucket;
+let user1;
+let user2;
+let user3;
+
+xdescribe('Test token bucket functionality', () => {
+    beforeAll(() => {
+        user1 = '1';
+        user2 = '2';
+        user3 = '3';
+    });
+
+    beforeEach(() => {
+        // Initialize a new token bucket before each test
+        // create a mock user
+        // intialze the token bucket algorithm
+        limiter = new TokenBucket(CAPACITY, REFILL_RATE);
+    });
+
+    test('allows a user to consume up to their current allotment of tokens', () => {
+        // Test 1 token requested
+        expect(limiter.processRequest(user1, 1)).toBe(true);
+        // Test < CAPACITY tokens requested
+        expect(limiter.processRequest(user2, CAPACITY - 1)).toBe(true);
+        // <= CAPACITY tokens requested
+        expect(limiter.processRequest(user3, CAPACITY)).toBe(true);
```

Comment:
For edge case: adding `expect(limiter.processRequest(user1, 0).toThrowError()` or something similar may be beneficial.

Reply:
I'll add a test for 0, but I think we should have the expected behavior in that case be to allow the request since it's "free". Thoughts?